dev_guide/builds.adoc

```diff
@@ -5,19 +5,40 @@
 :icons:
 :experimental:
 :toc: macro
+:toclevels: 3
 :toc-title:
 :prewrap!:
 
 toc::[]
 
 == Overview
-A link:../architecture/core_concepts/builds_and_image_streams.html#builds[build] is a process of creating
-runnable images to be used on OpenShift. There are three build strategies:
+
+A link:../architecture/core_concepts/builds_and_image_streams.html#builds[build]
+is a process of creating runnable images to be used on OpenShift.
+
+Build configurations are characterized by strategy and sources. The strategy
+determines the aforementioned process, while the sources provide its input.
+Every build configuration have a single strategy and at least one source.
+
+There are three build strategies:
 
 - link:../architecture/core_concepts/builds_and_image_streams.html#source-build[Source-To-Image (S2I)]
 - link:../architecture/core_concepts/builds_and_image_streams.html#docker-build[Docker]
 - link:../architecture/core_concepts/builds_and_image_streams.html#custom-build[Custom]
 
+And there are three types of build source:
+
+- link:#source-code[Git]
+- link:#dockerfile-source[Dockerfile]
+- link:#binary-source[Binary]
+
+It is up to each build strategy to consider or ignore a certain type of source,
+as well as to determine how it is to be used.
+
```

Comment:
Yeah, it was a hard time for me to digest all the nuances. @bparees helped me to arrive at the current stage.
Instead of a table I'd prefer just to have simpler "rules", and we're working on that.
That said, I think the table would be a poor time investment now.